### PR TITLE
Java native interface

### LINKED ISSUE #3521: Java Native Interface
### Proposed title of article
Java Native Interface

### Introduction paragraph (2-3 paragraphs):

As we know, one of Java's key advantages is its portability, which means that when we create and compile code, the outcome is platform-independent bytecode. Simply put, this can run on any machine or device that can run a Java Virtual Machine, and it will run as smoothly as possible.

However, there are situations when we need to use natively compiled code for a specific architecture. In this article the reader will be able to understand  Java Native Interface and how to implement it. 

### Key takeaways: 

   The reader will understand the following;

- Native Methods.
-  JNI Elements in Code (Java).
-  Creating the Java Class.
-  Compiling and Linking.
- Adding Parameters To Our Native Methods.
- Using Objects and Calling Java Methods From Native Code.
- Advantages and disadvantages of Using JNI.

### References: 
N/A

### PR DESCRIPTION
**Pull request by Shikomary**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3521 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Shikomary** [comment]
Hi, @BrianaNzivu 
Would you update me the progress of this PR please it has been quite some time with no updates.
Thank you.

**Comment by BrianaNzivu** [comment]
Hey @Shikomary.
Apologies for the delay.
Working on it.

**Comment by Shikomary** [comment]
Hello, @BrianaNzivu 
Aright thank you for the update

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @Shikomary 👋 .

First of all, welcome to Section - EngEd. :tada: 
We are happy to have you here.
Good job on the article. It is a very interesting read.
Just a few changes I need you to make as we continue with the preliminary revisions:
- Each article in Section has its own structure on how it is written. For instance, your article is more of a how-to guide. Please check out this [document](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md) and try and structure your article according to this guide.
- Your article has quite a lot of grammatical mistakes and unclear sentences. You can use 
[Grammarly](https://app.grammarly.com/ddocs/1287976180) to correct it. 
-Please correct the spelling for your avatar image. Have it as `avatar.png`.

In case of any question or help needed please do not hesitate to ping me here :smile:.
Also, notify me when you are done.

Regards.

**Comment by Shikomary** [comment]
Hi @BrianaNzivu 
I have done the changes you requested but am unable to commit the changes in my branch. I will be glad for your help.
Thank you.

**Comment by Shikomary** [comment]
Hi @BrianaNzivu 
I have done the changes that  you requested on the tutorial.
However am unable to change the name (avatar.png). Please guide me on that. 
Thank you.

**Comment by BrianaNzivu** [comment]
> Hi @BrianaNzivu 
> I have done the changes that  you requested on the tutorial.
> However am unable to change the name (avatar.png). Please guide me on that. 
> Thank you.

Hey @Shikomary .
You can delete the current one on the PR and upload a new one.

**Comment by Shikomary** [comment]
Hi, @BrianaNzivu 
I have done all the changes.
Thank You.

**Comment by BrianaNzivu** [comment]
Still seeing two avatar images. Please delete the `avatar.png.jpg` one.

File: content/authors/mary-wanjiku/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: Mary Wanjiku
+type: authors
```
Comment (BrianaNzivu) [inline content/authors/mary-wanjiku/index.md:3]: Please do not make changes to the author bio.

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

**Comment by Shikomary** [comment]
Hi, @BrianaNzivu 
I have changed the name of the author as it was initially.  You can update me on any issues you get.
Thank you.

**Comment by BrianaNzivu** [comment]
> Hi, @BrianaNzivu 
> I have changed the name of the author as it was initially.  You can update me on any issues you get.
> Thank you.

It's ok.
I still need you to delete the extra avatar image...
The one labeled avatar.png.jpg

**Comment by Shikomary** [comment]
hello, @BrianaNzivu 
I have deleted the avatar image.
Thank you.

**Comment by BrianaNzivu** [comment]
Tech review done 👍

**Review by BrianaNzivu — APPROVED** [review]
Hey @FranciscaNg  👋 .
The article is a very interesting read. Good job!
Just a few things I'd like to point out:
1.  It is always advisable to submit an article that is well polished. By this, I mean having an article that does not have too many grammatical corrections and functional code. I suggest that you check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) page which has links to various resourceful tools for writing. 
2. Lastly please go through all the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your first article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

**Comment by Shikomary** [comment]
Hi, @BrianaNzivu 
Thank you for reviewing my PR and for your guide. 
I appreciate that.

**Comment by Kayere** [comment]
Congratulations for publishing your first article @Shikomary. Please go over the changes made by your peer reviewer and I to help improve the quality of your future articles.

Otherwise, awesome work done. We hope this is the first of many quality articles 🚀.

Thank you for the review @BrianaNzivu 🚀.

**Pull request merged**